api/restHandler/PipelineConfigRestHandler.go

```diff
@@ -1234,26 +1250,42 @@ func (handler PipelineConfigRestHandlerImpl) UpdateAppOverride(w http.ResponseWr
 		return
 	}
 	handler.Logger.Infow("request payload, UpdateAppOverride", "payload", templateRequest)
-	token := r.Header.Get("token")
-	app, err := handler.pipelineBuilder.GetApp(templateRequest.AppId)
-	if err != nil {
-		writeJsonResp(w, err, nil, http.StatusBadRequest)
-		return
+	buff, merr := json.Marshal(templateRequest)
+	if merr != nil {
+		handler.Logger.Errorw("marshal err, handleForwardResponseStreamError", "err", merr, "response", templateRequest)
 	}
 
-	resourceName := handler.enforcerUtil.GetAppRBACName(app.AppName)
-	if ok := handler.enforcer.Enforce(token, rbac.ResourceApplications, rbac.ActionCreate, resourceName); !ok {
-		writeJsonResp(w, fmt.Errorf("unauthorized user"), "Unauthorized User", http.StatusForbidden)
-		return
+	var dat pipeline.TemplateRequest
+
+	if err := json.Unmarshal(buff, &dat); err != nil {
+		panic(err)
 	}
+	chartVersion := dat.RefChartTemplate
+	if validatejson(dat, chartVersion) {
```

Comment:
shouldn't we validate after authorization?

Reply:
Done Sir.